### DISCUSSION TITLE
Run Renovate on Demand when clicking on Rebase PR

### DISCUSSION THREAD

**Discussion by rajat053117** [renovatebot/renovate#30009] [Suggest an Idea]
### Tell us more.

Subject: Proposal to Enhance Renovate PR Rebase Functionality

Dear Renovate Team,

First, I would like to extend my heartfelt gratitude for the marvellous Renovate app and the incredible support you provide. Your assistance has been invaluable in resolving our issues, and for that, we are deeply thankful.

I am writing to propose an enhancement to the Renovate app that I believe would significantly improve its usability and efficiency for many users.

Currently, we run Renovate in a centralized repository, and it executes on a scheduled basis every four hours. During each run, it fetches dependencies as specified in each repository's configuration file. While this setup works well for most cases, there is one area where we encounter inefficiencies: the rebasing of pull requests (PRs).

When Renovate creates a PR, there is an option to rebase the PR. However, clicking this option only triggers the rebase during the next scheduled run. This delay can be problematic as it prevents us from merging the PR promptly, often leaving us waiting idly or forcing us to run Renovate manually via GitHub Actions.

To address this, I propose an enhancement: when the "Rebase PR" button is clicked on a PR generated by Renovate, it should automatically trigger the Renovate GitHub Action from our centralized repository specifically for that repository and rebase the PR immediately.

This functionality would greatly enhance our workflow by allowing more timely merges of PRs, reducing downtime, and improving overall efficiency.

Is this enhancement possible? If so, what steps would be necessary to implement this feature?

**Comment by rarkins** [comment]
This is not possible, because GitHub Actions do not have webhook listeners. You can achieve this instead using Renovate CE or EE: https://github.com/mend/renovate-ce-ee

**Comment by jamietanna** [comment]
You may be able to hack something together with the `issues: edited` [event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues) to make this work, but I'd +1 that using Mend Renovate CE is a much better solution + means you don't need to think about setting this up, you can just get going with that :rocket:
